Source/Model/ModelParser.cs

```diff
@@ -308,7 +310,8 @@ internal override void Run()
                 continue;
               }
 
-              var args = new Model.Element[fn.Arity];
+              fn.Arity = tuple.Count - 2;
+              var args = new Model.Element[fn.Arity ?? 1]; // arity cannot be null here
```

Comment:
If Arity cannot be null, don't you want new Model.Element[fn.Arity]?

Reply:
You can cast (int) fn.Arity.

Reply:
Right, thank you! Not sure why I did it that way. Fixed that in the new commit.